### ISSUE TITLE
[Feedback needed] What do you use Sidekick for?

### ISSUE THREAD

**Issue by MightyMoud** [MightyMoud/sidekick#30]
Hi guys,

I have outlined my vision for Sidekick in the Readme. You can have a look if you haven't read it yet. 
However, from reading the issues in the past couple of weeks I can't help but see many people use it for different things; homelabs, automation and so on. 

As I think about the future of this project it would be very helpful to hear from you guys how do you use or intend to use Sidekick? 

Would be very grateful for detailed answers.

Cheers,
MM

**Comment by daninus14** [comment]
Thank you for sharing. This looks amazing. I intent on using it to deploy web applications, so the feature to set up a postgres DB easily would really be appreciated 😄

**Comment by MightyMoud** [comment]
> Thank you for sharing. This looks amazing. I intent on using it to deploy web applications, so the feature to set up a postgres DB easily would really be appreciated 😄

Yeah that's planned. Once the core is stable, I will start on adding db support

**Comment by PKuebler** [comment]
I deploy all private projects with Docker. To do this, I run linter and tests in the pipeline in Github/Gitlab and if these are successful, the pipeline deploys the Docker container via SSH/Bash. Would like to leave the deploying from the pipeline to sidekick. Because branches as a feature preview would be very cool. For example, I have a preview stage for my blog to test new features.

**Comment by rubiojr** [comment]
I currently setup my VPS servers manually with docker-compose+Caddy and bind mounted directories as data volumes to host any kind of app, self made and otherwise. So very similar to a Sidekick setup. Some examples of third party apps I use:

- Pocketbase - https://github.com/pocketbase/pocketbase
- Photoview - https://photoview.github.io

I've been evaluating Sidekick to replace my custom VPS builds with success so far. Sidekick philosophy really resonates with me, tired of the complexity of modern deploy strategies.

Things I'm looking for:

* Painless 1 command deploy with TLS. This is where Sidekick already excels for certain class of apps 🏆 .
* Backups are essential for anything I host, and backing up volumes associated volumes of a containers isn't straight forward with Docker out of the box, when using docker native volumes (reason I bind mount directories as volumes). If I could `sidekick backup` and that would drop a `.tgz` per container volume somewhere, that'd be a blast.
* Simplifying the data volumes story. It'd be nice if Sidekick made hosting data intensive apps that don't rely on third party storage APIs (i.e. local storage only) easier.
* I script my app's FQDN creation with `flarectl` (Cloudflare's), before creating the containers, so the domain is ready for the container at container creation time. Cloudflare (or other similar service) integration to create the 's FQDN while `sidekick launch` is run (creating the FQDN automatically), would be a blast.

Congratulations, Sidekick's future is looking really bright 🌞.

**Comment by apertureless** [comment]
Really like the vision / idea of sidekick. Would love to use it to deploy client projects.

However, coming also from a security background, reading "production ready" vps and seeing no server hardening whatsoever kinda leaves a weird feeling.

Would be awesome if the provisioned server would follow some best practices for hardening like [CIS](https://www.cisecurity.org/cis-benchmarks) or [NIST](https://ncp.nist.gov/repository) for general system hardening and service hardening (ssh, etc)

**Comment by hlibco** [comment]
Currently using Sidekick to deploy web apps (Go and Typescript + Bun). Databases and Queues I generally use cloud-managed. I appreciate the vision around simplicity and wizard vs magic. A few nice to haves would be:

- `hooks` (pre / post "launch" and "deploy")
- New commands: `list` (list all deployed apps), `stop` (which would stop traffic without removing the container and deleting the image) and `start` (which would start serving traffic for stopped apps).
